### PR TITLE
Utility for collecting stats about ixTheo notations in MARC data sets.

### PR DESCRIPTION
**Pull request by ruschein**

### CODE REVIEW

File: cpp/ixtheo_notation_stats.cc [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+#include "util.h"
+
+
+void Usage() {
+    std::cerr << "Usage: " << progname << "marc_input code_to_description_map\n";
+    std::exit(EXIT_FAILURE);
+}
+
+
+void LoadCodeToDescriptionMap(const std::shared_ptr<FILE> &code_to_description_map_file,
```
Comment (oobenland) [inline cpp/ixtheo_notation_stats.cc]: why do you use this list? Why didn't you simplify the syntax of the list, something like
AB=Religionswissenschaft
The parsing of this function is very confusing.

File: cpp/ixtheo_notation_stats.cc [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+        ++count;
+
+        if (dir_entries[0].getTag() != "001")
+            Error("First field is not \"001\"!");
+
+	int lok_index(MarcUtil::GetFieldIndex(dir_entries, "LOK"));
+	bool found_at_least_one(false);
+	while (lok_index != -1) {
+	    const Subfields subfields(field_data[lok_index]);
+	    const auto begin_end(subfields.getIterators('0'));
```
Comment (oobenland) [inline cpp/ixtheo_notation_stats.cc]: this is a bad name. Maybe subfield_zero_iterator?

File: cpp/ixtheo_notation_stats.cc [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+	    if (std::distance(begin_end.first, begin_end.second) == 2 and begin_end.first->second == "936ln") {
```
Comment (oobenland) [inline cpp/ixtheo_notation_stats.cc]: You should use `begin_end.end` here. Even thoug you get the result of std::distance` right, it dosn't check if there are only two elements.

File: cpp/ixtheo_notation_stats.cc (hunk 1 above)
Comment (ruschein) [inline cpp/ixtheo_notation_stats.cc]: Good idea!  Done!

**Pull request merged**